packages/user_packages/iotcc_mqtt.py

```diff
@@ -0,0 +1,94 @@
+#      documentation and/or other materials provided with the distribution.   #
+#                                                                             #
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"#
+#  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE  #
+#  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE #
+#  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE  #
+#  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR        #
+#  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF       #
+#  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS   #
+#  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN    #
+#  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)    #
+#  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF     #
+#  THE POSSIBILITY OF SUCH DAMAGE.                                            #
+# ----------------------------------------------------------------------------#
+
+from liota.core.package_manager import LiotaPackage
+
+dependencies = ["general_edge_system", "credentials"]
+
+
+class PackageClass(LiotaPackage):
+    """
+    This is a sample package which creates a IoTControlCenter DCC object and registers edge system on
+    IoTCC over MQTT Protocol to acquire "registered edge system", i.e. iotcc_edge_system.
+    """
+
+    def run(self, registry):
+        import copy
+        from liota.lib.utilities.identity import Identity
+        from liota.dccs.iotcc import IotControlCenter
+        from liota.dcc_comms.mqtt_dcc_comms import MqttDccComms
+        from liota.lib.transports.mqtt import MqttMessagingAttributes
+        from liota.dccs.dcc import RegistrationFailure
+        from liota.lib.utilities.tls_conf import TLSConf
+
+        # Acquire resources from the registry
+        # Creating a copy of edge_system object to keep original object "clean"
+        edge_system = copy.copy(registry.get("edge_system"))
+
+        #  Encapsulates Identity
+        # Acquire credentials and required certificates from the registry
+        identity = Identity(root_ca_cert=registry.get("broker_root_ca_cert"), username=registry.get("broker_username"),
+                            password=registry.get("broker_password"),
+                            cert_file=registry.get("edge_system_cert_file"), key_file=registry.get("edge_system_key_file"))
+
+        # Encapsulate TLS parameters
+        tls_conf = TLSConf(cert_required="CERT_REQUIRED", tls_version="PROTOCOL_TLSv1_2", cipher=None)
+
+        # Initialize DCC object with MQTT transport
+        mqtt_msg_attr = MqttMessagingAttributes(pub_topic="liota/"+registry.get("broker_username")+"/request",sub_topic="liota/"+registry.get("broker_username")+"/response")
+        self.iotcc = IotControlCenter(MqttDccComms(edge_system_name=edge_system.name,
```

Comment:
Now, rather than let MqttDccComms automatically generate pub/sub topics, you choose to generate unique pub/sub topics based on "broker_username", right?
credentials liota package will be inserted during bootstrap package creation through UI, right?
ACL then must be configured on mqtt broker to create unique broker_username per client, right?
O.w., same pub/sub topics could be used by multiple clients. Shouldn't we point this out to let users be aware?